core/src/components/modal/modal.ios.scss

```diff
@@ -5,6 +5,9 @@
 // --------------------------------------------------
 
 :host {
+  --ion-background-color: var(--ion-components-ion-modal-bg);
+  --ion-toolbar-background: var(--ion-components-ion-modal-toolbar-background);
+  --ion-toolbar-border-color: var(--ion-components-ion-modal-toolbar-border-color);
```

Comment:
These were added due to modal having to change variables based on [theming](https://github.com/ionic-team/ionic-framework/pull/30799/files#diff-e9e035eab6ff006fb7bdb9bb98162f6457326203037cc9f209d76579ad6dc013L142-L146).